sdk/src/main/java/com/mapbox/maps/MapController.kt

```diff
@@ -247,44 +253,6 @@ internal class MapController : MapPluginProviderDelegate, MapControllable {
     renderer.renderThread.setScreenRefreshRate(refreshRate)
   }
 
-  //
-  // Telemetry
-  //
-
-  private fun dispatchTelemetryTurnstileEvent(): MapTelemetry {
-    val telemetry: MapTelemetry =
-      MapboxModuleProvider.createModule(MapboxModuleType.MapTelemetry) {
-        paramsProvider(MapboxModuleType.MapTelemetry)
-      }
-    telemetry.onAppUserTurnstileEvent()
-    return telemetry
-  }
-
-  /**
-   * Provides parameters for Mapbox default modules, recursively if a module depends on other Mapbox modules.
-   */
-  private fun paramsProvider(type: MapboxModuleType): Array<ModuleProviderArgument> {
-    val context = mapInitOptions.context
-    return when (type) {
-      MapboxModuleType.CommonLibraryLoader -> arrayOf(
-        ModuleProviderArgument(
-          Context::class.java,
-          context.applicationContext
-        )
-      )
-      MapboxModuleType.CommonHttpClient -> arrayOf()
```

Comment:
Wonder if we should also proceed and remove the module system completely?

Reply:
Also wonder if we don't fully remove the module setup, and there're still other modules in the codebase, we might still encounter the similar issue in other code paths, e.g https://github.com/mapbox/mapbox-maps-android/issues/1540

Reply:
> As far as I know common SDK still uses the `CommonHttpClient` module setup, but seems they have their own `paramsProvider`.

Right - but if there is no clear reasons to do so it will be nice to align and avoid exact issues that are still possible with reflection

Reply:
The module setup is widely used across common and nav sdks, it exists for the flexibility to let the user to provide their own overwrites to some functionalities. And we have [this](https://docs.mapbox.com/android/maps/guides/migrate-to-v10/#example-replace-modular-http-stack) section in the migration guide to use this system to overwrite the http stack. If we want to totally remove the modular setup, it would be a bigger change, and probably in v11.

Reply:
Hmm, indeed, then suppose we need to keep the public API for those users. 
Can we though somehow modify the way Modules are generated so that it wouldn't involve reflection? Should be possible, e.g. add some setters to the [CommonModuleProvider](https://github.com/mapbox/mapbox-sdk-common/blob/97257fc308917dd07a4568847b87f1fe201958c6/platform/android/common/src/main/java/com/mapbox/common/core/module/CommonSingletonModuleProvider.kt#L15) and call them from the generated modules